### PR TITLE
Auto-detect update_jacobian_grads

### PR DESCRIPTION
**Pull request by NiklasWik**
Follow-up to #13990.

If gradients are requested with `FE_RaviartThomasNodal` on non-affine cells we also need the jacobian gradient.

### CODE REVIEW

**Review by kronbichler — APPROVED** [review]

File: include/deal.II/matrix_free/mapping_info_storage.templates.h [hunk 1]
```diff
@@ -135,7 +136,10 @@ namespace internal
       // Jacobians (these two together will give use the gradients of the
       // inverse Jacobians, which is what we need)
       if ((update_flags & update_hessians) != 0u ||
-          (update_flags & update_jacobian_grads) != 0u)
+          (update_flags & update_jacobian_grads) != 0u ||
```
Comment (peterrum) [inline include/deal.II/matrix_free/mapping_info_storage.templates.h:139]: Do we still have these (`!= 0u`)?

File: include/deal.II/matrix_free/mapping_info.h [hunk 2]
```diff
@@ -74,7 +74,8 @@ namespace internal
         const UpdateFlags update_flags_cells,
         const UpdateFlags update_flags_boundary_faces,
         const UpdateFlags update_flags_inner_faces,
-        const UpdateFlags update_flags_faces_by_cells);
+        const UpdateFlags update_flags_faces_by_cells,
+        const bool        piola_transform = false);
```
Comment (peterrum) [inline include/deal.II/matrix_free/mapping_info.h:78]: Could we be explicit here?
```suggestion
        const bool        piola_transform);
```

File: include/deal.II/matrix_free/matrix_free.templates.h [hunk 3]
```diff
@@ -35,6 +35,7 @@
 #include <deal.II/fe/fe_dgq.h>
 #include <deal.II/fe/fe_poly.h>
 #include <deal.II/fe/fe_q_dg0.h>
+#include <deal.II/fe/fe_raviart_thomas.h>
```
Comment (peterrum) [inline include/deal.II/matrix_free/matrix_free.templates.h:38]: Why is this include needed here?

File: include/deal.II/matrix_free/mapping_info_storage.templates.h (hunk 1 above)
Comment (NiklasWik) [reply include/deal.II/matrix_free/mapping_info_storage.templates.h:139]: In quite a few places in `include/matrix_free/evaluation_kernels.h` as well.

File: include/deal.II/matrix_free/matrix_free.templates.h (hunk 3 above)
Comment (NiklasWik) [reply include/deal.II/matrix_free/matrix_free.templates.h:38]: Forgot to remove it as I first did some `dynamic_cast` as in `shape_info.templates.h` but figured that I didn't have to do it twice! Thanks

**Review by kronbichler — APPROVED** [review]

**Pull request merged**